pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableInstances.java

```diff
@@ -116,16 +121,42 @@ public String getTableInstances(
   }
 
   @GET
-  @Path("/tables/{tableNameWithType}/livebrokers")
+  @Path("/tables/{tableName}/livebrokers")
   @Produces(MediaType.APPLICATION_JSON)
   @ApiOperation(value = "List the brokers serving a table", notes = "List live brokers of the given table based on EV")
   @ApiResponses(value = {
       @ApiResponse(code = 200, message = "Success"),
       @ApiResponse(code = 404, message = "Table not found"),
       @ApiResponse(code = 500, message = "Internal server error")})
   public List<String> getLiveBrokersForTable(
-      @ApiParam(value = "Table name with type", required = true)
-      @PathParam("tableNameWithType") String tableNameWithType) {
-    return _pinotHelixResourceManager.getLiveBrokersForTable(tableNameWithType);
+      @ApiParam(value = "Table name (with or without type)", required = true)
+      @PathParam("tableName") String tableName) {
+    if (TableNameBuilder.getTableTypeFromTableName(tableName) != null) {
+      return _pinotHelixResourceManager.getLiveBrokersForTable(tableName);
```

Comment:
Shall we push down this logic into the `PinotHelixResourceManager`? I.e. change `PinotHelixResourceManager.getLiveBrokersForTable()` to take table name with or without type suffix

Reply:
I see that some methods in `PinotHelixResourceManager` use `tableName` and some use `tableNameWithType` so in that sense it should be fine. Pushing it down to `PinotHelixResourceManager` would allow other callers of this method to avoid having to duplicate the logic the API is doing, so that's a positive. I'll update.